lib/experimental/class-theme-json-webfonts-handler.php

```diff
@@ -0,0 +1,537 @@
+				}
+
+				return $css;
+			}
+
+			/**
+			 * Order `src` items to optimize for browser support.
+			 *
+			 * @since 6.0.0
+			 *
+			 * @param array $webfont Webfont to process.
+			 *
+			 * @return array Ordered `src` items.
+			 */
+			private function order_src( array $webfont ) {
+				$src         = array();
+				$src_ordered = array();
+
+				foreach ( $webfont['src'] as $url ) {
+					// Add data URIs first.
+					if ( 0 === strpos( trim( $url ), 'data:' ) ) {
+						$src_ordered[] = array(
+							'url'    => $url,
+							'format' => 'data',
+						);
+						continue;
+					}
+					$format         = pathinfo( $url, PATHINFO_EXTENSION );
+					$src[ $format ] = $url;
+				}
+
+				// Add woff2.
+				if ( ! empty( $src['woff2'] ) ) {
+					$src_ordered[] = array(
+						'url'    => $src['woff2'],
+						'format' => 'woff2',
+					);
+				}
+
+				// Add woff.
+				if ( ! empty( $src['woff'] ) ) {
+					$src_ordered[] = array(
+						'url'    => $src['woff'],
+						'format' => 'woff',
+					);
+				}
+
+				// Add ttf.
+				if ( ! empty( $src['ttf'] ) ) {
+					$src_ordered[] = array(
+						'url'    => $src['ttf'],
+						'format' => 'truetype',
+					);
+				}
+
+				// Add eot.
+				if ( ! empty( $src['eot'] ) ) {
+					$src_ordered[] = array(
+						'url'    => $src['eot'],
+						'format' => 'embedded-opentype',
+					);
+				}
+
+				// Add otf.
+				if ( ! empty( $src['otf'] ) ) {
+					$src_ordered[] = array(
+						'url'    => $src['otf'],
+						'format' => 'opentype',
+					);
+				}
+				$webfont['src'] = $src_ordered;
+
+				return $webfont;
+			}
+
+			/**
+			 * Builds the font-family's CSS.
+			 *
+			 * @since 6.0.0
+			 *
+			 * @param array $webfont Webfont to process.
+			 *
+			 * @return string This font-family's CSS.
+			 */
+			private function build_font_face_css( array $webfont ) {
+				$css = '';
+
+				// Wrap font-family in quotes if it contains spaces.
+				if (
+					str_contains( $webfont['font-family'], ' ' ) &&
+					! str_contains( $webfont['font-family'], '"' ) &&
+					! str_contains( $webfont['font-family'], "'" )
+				) {
+					$webfont['font-family'] = '"' . $webfont['font-family'] . '"';
+				}
+
+				foreach ( $webfont as $key => $value ) {
+					/*
+					 * Skip "provider", since it's for internal API use,
+					 * and not a valid CSS property.
+					 */
+					if ( 'provider' === $key ) {
+						continue;
+					}
+
+					// Compile the "src" parameter.
+					if ( 'src' === $key ) {
+						$value = $this->compile_src( $webfont['font-family'], $value );
+					}
+
+					// If font-variation-settings is an array, convert it to a string.
+					if ( 'font-variation-settings' === $key && is_array( $value ) ) {
+						$value = $this->compile_variations( $value );
+					}
+
+					if ( ! empty( $value ) ) {
+						$css .= "$key:$value;";
+					}
+				}
+
+				return $css;
+			}
+
+			/**
+			 * Compiles the `src` into valid CSS.
```

Comment:
```suggestion
			 * Compiles the 'src' into valid CSS.
```
- Officially, markdown syntax is only valid in the body of a docblock, not the title.